src/main/java/com/mycompany/myapp/web/rest/reservation/ReservationResource.java

```diff
@@ -0,0 +1,55 @@
+package com.mycompany.myapp.web.rest.reservation;
+
+import com.codahale.metrics.annotation.Timed;
+import com.mycompany.myapp.domain.Reservation;
+import com.mycompany.myapp.service.reservation.ReservationService;
+import com.mycompany.myapp.web.rest.util.HeaderUtil;
+import io.github.jhipster.web.util.ResponseUtil;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/reservation")
+public class ReservationResource {
+
+    ReservationService reservationService;
+
+    public ReservationResource(ReservationService reservationService){
+        this.reservationService = reservationService;
+    }
+
+    /**
+     * GET  /reserve : make reservation for pool
+     *
+     */
+    @GetMapping("/reserve")
```

Comment:
I kinda don't understand this code.

HTTP GET suggest fetching but you return `Void` and then generate `deleted` label. Also name suggests making the reservation.

Reply:
actually there is nothing wrong with get request that save smth and return nothing, it is kind of sad practice I agree, but people tend to do it. I shouldn't generate delete label tho